internal/handlers/gateway/tempo_secert.go

```diff
@@ -0,0 +1,75 @@
+package gateway
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/os-observability/tempo-operator/apis/tempo/v1alpha1"
+	"github.com/os-observability/tempo-operator/internal/manifests/manifestutils"
+	"github.com/os-observability/tempo-operator/internal/status"
+)
+
+// GetTenantSecrets returns the list to gateway tenant secrets for a tenant mode.
+// For modes static the secrets are fetched from external provided secrets.
+// All secrets live in the same namespace as the tempostack request.
+func GetTenantSecrets(
+	ctx context.Context,
+	k client.Client,
+	req ctrl.Request,
```

Comment:
why the `req` is passed? Only namespace is used from it. Isn't the namespace already set in the `stack` variable?